tests/unit/peapods/runtimes/asyncio/rest/test_models.py

```diff
@@ -73,45 +69,50 @@ def test_all_fields_in_document_proto():
     ]:
         assert i in document_proto_properties
 
-    document_proto_properties_alias = PROTO_TO_PYDANTIC_MODELS[
-        'DocumentProto'
-    ]().schema()['definitions']['DocumentProto']['properties']
+    document_proto_properties_alias = PROTO_TO_PYDANTIC_MODELS.DocumentProto().schema()[
+        'definitions'
+    ]['DocumentProto']['properties']
     for i in ['contentHash', 'levelName', 'parentId', 'mimeType']:
         assert i in document_proto_properties_alias
 
 
 def test_oneof():
     """ This tests: oneof field is correctly represented as `anyOf` """
-    content = PROTO_TO_PYDANTIC_MODELS['DocumentProto']().schema()['definitions'][
-        'DocumentProto'
-    ]['properties']['content']
-    assert 'anyOf' in content
-    assert len(content['anyOf']) == 3
-    assert {'$ref': '#/definitions/NdArrayProto'} in content['anyOf']
 
-    body = PROTO_TO_PYDANTIC_MODELS['RequestProto']().schema()['properties']['body']
-    assert 'anyOf' in body
-    assert len(body['anyOf']) == 6
-    assert {'$ref': '#/definitions/IndexRequestProto'} in body['anyOf']
-    assert {'$ref': '#/definitions/TrainRequestProto'} in body['anyOf']
+    doc = PROTO_TO_PYDANTIC_MODELS.DocumentProto(text='abc')
+    assert doc.text == 'abc'
+    assert 'blob' not in doc.dict()
+    assert 'buffer' not in doc.dict()
+
+    doc = PROTO_TO_PYDANTIC_MODELS.DocumentProto(buffer=b'abc')
+    assert doc.buffer == b'abc'
+    assert 'text' not in doc.dict()
+    assert 'blob' not in doc.dict()
+
+    doc = PROTO_TO_PYDANTIC_MODELS.DocumentProto(
+        blob=PROTO_TO_PYDANTIC_MODELS.NdArrayProto()
+    )
+    assert doc.blob == PROTO_TO_PYDANTIC_MODELS.NdArrayProto()
+    assert 'text' not in doc.dict()
+    assert 'buffer' not in doc.dict()
```

Comment:
Can we add a test for the content attribute as well?
```
    doc = PROTO_TO_PYDANTIC_MODELS.DocumentProto(content=b'abc')
    assert doc.buffer == b'abc'
    assert 'text' not in doc.dict()
    assert 'blob' not in doc.dict()
    ```
    
    The document creation works the same way:
    
Document(content='my text')
<jina.types.document.Document id=8a2788b2-823f-11eb-b8b8-faffc24bcbdb mimeType=text/plain text=my text at 140330282190992>

Reply:
- `PROTO_TO_PYDANTIC_MODELS` is trying to replicate every proto definition in `jina.proto`, but not things in `Document()`

```python
from jina.proto.jina_pb2 import DocumentProto
DocumentProto(content='abc')
# ValueError: Protocol message DocumentProto has no "content" field.
# `oneof` key 'content' cannot be set

DocumentProto(text='abc').content
# AttributeError: content
# `oneof` key 'content' doesn't get set if we set the oneof fields

from jina.peapods.runtimes.asyncio.rest.models import PROTO_TO_PYDANTIC_MODELS
PROTO_TO_PYDANTIC_MODELS.DocumentProto(text='abc').content
# AttributeError: 'DocumentProto' object has no attribute 'content'
# Like proto, we too don't set content 
```

- `jina.types.document.Document()` can read from the proto directly or `bytes` or `str` or `Dict`. Pydantic models help us send the doc as str or dict to `Document()` for REST based requests, I should add tests for the below.

```python
DocumentSourceType = TypeVar(
    'DocumentSourceType', jina_pb2.DocumentProto, bytes, str, Dict
)

str -> PROTO_TO_PYDANTIC_MODELS.DocumentProto().json()
Dict -> PROTO_TO_PYDANTIC_MODELS.DocumentProto().dict()
```
- The oneof key setting happens in `Document()` for both proto & pydantic model.
```python
proto_doc = DocumentProto(text='abc')
Document(proto_doc).content
# 'abc'

pydantic_doc_dict = PROTO_TO_PYDANTIC_MODELS.DocumentProto(text='abc').dict()
Document(pydantic_doc_dict).content
# 'abc'
```

Reply:
@hanxiao please let me know if this assumption is not right.